test/index.js

```diff
@@ -2,33 +2,88 @@ const lkl = require('../');
 const fs = require('fs');
 const path = require('path');
 const assert = require('assert');
+const async = require('async');
+const zlib = require('zlib');
 
 let mountpoint;
 
 describe('lkl', function() {
 	before(function(done) {
-		this.disk = __dirname + '/tmp.ext4';
-
+		const self = this;
+		this.disk = __dirname + '/tmp.ext4';  // ext4 partition
+		this.partitionnedDisk = __dirname + '/tmp.img';  // whole disk
+		this.partitions = [  // whole disk partitions
+			{ number: 1, fsType: 'btrfs' },
+			{ number: 2, fsType: 'ext2' },
+			{ number: 3, fsType: 'ext4' },
+			// 4 is an extended partition
+			{ number: 5, fsType: 'vfat' },  // fat16
+			{ number: 6, fsType: 'vfat' },  // fat32
+			{ number: 7, fsType: 'xfs' },
+		]
+
+		// create a copy of the ext4 partition
 		fs.createReadStream(__dirname + '/fixtures/test.ext4')
 		.pipe(fs.createWriteStream(this.disk))
-		.on('close', done);
+		.on('close', function() {
+			// then create a copy of the partitionned disk.
+			// The disk image is compressed in order to avoid github's file
+			// size restrictions.
+			fs.createReadStream(__dirname + '/fixtures/disk.img.gz')
```

Comment:
I wanted to test that we can mount any file system.
All the partitions have the minimum required size for each FS:


| number | file system | size   |
|--------|-------------|--------|
| 1      | btrfs       | 256Mio |
| 2      | ext2        | 1Mio   |
| 3      | ext4        | 1Mio   |
| 4      | extended    |        |
| 5      | fat16       | 16Mio  |
| 6      | fat32       | 33Mio  |
| 7      | xfs         | 32Mio  |

This totals to 339Mio ~= 355MB

The compressed image is less than 427Kib and the decompression takes less than 2 seconds so I didn't think it was a problem.

We could remove the btrfs one and save 256 uncompressed Mio.